app-drawer-layout/app-drawer-layout.html

```diff
@@ -223,46 +216,31 @@
         }
       },
 
-      resetLayout: function() {
-        this.debounce('_resetLayout', function() {
-          var drawer = this.drawer;
-          var contentContainer = this.$.contentContainer;
+      _updateLayoutStates: function() {
+        var drawer = this.drawer;
+        var contentContainer = this.$.contentContainer;
 
-          if (this.narrow || !drawer) {
+        if (this.narrow || !drawer || !drawer.getWidth) {
+          contentContainer.style.marginLeft = '';
+          contentContainer.style.marginRight = '';
+        } else {
+          var drawerWidth = drawer.getWidth();
+          if (drawer.position == 'right') {
             contentContainer.style.marginLeft = '';
-            contentContainer.style.marginRight = '';
+            contentContainer.style.marginRight = drawerWidth + 'px';
           } else {
-            var drawerWidth = drawer.getWidth();
-            if (drawer.position == 'right') {
-              contentContainer.style.marginLeft = '';
-              contentContainer.style.marginRight = drawerWidth + 'px';
-            } else {
-              contentContainer.style.marginLeft = drawerWidth + 'px';
-              contentContainer.style.marginRight = '';
-            }
-          }
-        });
-      },
-
-      _resetDrawerState: function() {
```

Comment:
@frankiefu `_resetLayout` and `_resetDrawerState` were separated when I added `openedWhenNarrow` (which allows you to keep the drawer open on initial load/when the viewport becomes narrow). We only want to change the drawer opened state when `narrow` changes, not on every window resize event, so that's why it's separate.

This is why we have `app-drawer-attached`, which only fires once when attached so the `opened` state can be set then.